src/plugins/es_ui_shared/__packages_do_not_import__/authorization/components/with_privileges.tsx

```diff
@@ -26,24 +27,30 @@ interface Props {
 
 type Privilege = [string, string];
 
-const toArray = (value: string | string[]): string[] =>
+const toArray = (value: Privileges): string[] =>
   Array.isArray(value) ? (value as string[]) : ([value] as string[]);
 
-export const WithPrivileges = ({ privileges: requiredPrivileges, children }: Props) => {
-  const { isLoading, privileges } = useAuthorizationContext();
-
-  const privilegesToArray: Privilege[] = toArray(requiredPrivileges).map((p) => {
-    const [section, privilege] = p.split('.');
+export const convertPrivilegesToArray = (privileges: Privileges): Privilege[] => {
```

Comment:
I patched this block of code to also accept privileges that might contain dots in the name. I extracted this part into its own function and wrote a few tests for it, I decided not to write tests for the HOC itself because it seemed a bit outside the scope of this PR.